src/intops/impl/inlineasm/x86.nim

```diff
@@ -0,0 +1,161 @@
+## Arithmetic operations for integers implemented in x86 GCC/Clang inline Assembly.
+
+import ../../consts
+
+when cpu64Bit and cpuX86 and compilerGccCompatible and canUseInlineAsm:
+  func carryingAdd*(a, b: uint64, carryIn: bool): (uint64, bool) {.inline.} =
+    var
+      sum = a
+      cOut: uint8
```

Comment:
```cOut {.noinit.}: uint8```

`{.noinit.}` pragma at the end of function only affecting return value. `{.noninit.}` of variable declaration works for individual variable.